connect/runtime/src/main/java/org/apache/kafka/connect/util/KafkaBasedLog.java

```diff
@@ -312,6 +313,10 @@ public void run() {
                         try {
                             readToLogEnd();
                             log.trace("Finished read to end log for topic {}", topic);
+                        } catch (TimeoutException e) {
```

Comment:
there are a couple of other places where we call `readToEnd()` and `poll()`  (links below). should we catch this exception in those places also?

https://github.com/apache/kafka/blob/176ea0d0f9b4386cdc0e3b9bab560a11e3d3d516/connect/runtime/src/main/java/org/apache/kafka/connect/util/KafkaBasedLog.java#L332

https://github.com/apache/kafka/blob/176ea0d0f9b4386cdc0e3b9bab560a11e3d3d516/connect/runtime/src/main/java/org/apache/kafka/connect/util/KafkaBasedLog.java#L153

Reply:
Thanks for the additional eyes @wicknicks !

The `readToLogEnd()` call is inside `start()`so my thinking was that failure would happen on startup which seems okay (if not desirable).  If there were broker availability issues it seems reasonable to me for Connect not to start.  If we want to try to be more robust we could come up with some retrying strategy, but I bet that would require changes in a number of other places.

The `poll()` method contains its own `catch (KafkaException e)` which is a super class of `TimeoutException` so we ought to be okay there:
https://github.com/apache/kafka/blob/trunk/connect/runtime/src/main/java/org/apache/kafka/connect/util/KafkaBasedLog.java#L267